tests/components/bluetooth/test_passive_update_coordinator.py

```diff
@@ -0,0 +1,422 @@
+        ),
+    },
+)
+
+
+async def test_basic_usage(hass):
+    """Test basic usage of the PassiveBluetoothDataUpdateCoordinator."""
+
+    @callback
+    def _async_generate_mock_data(
+        service_info: BluetoothServiceInfo,
+    ) -> PassiveBluetoothDataUpdate:
+        """Generate mock data."""
+        return GENERIC_PASSIVE_BLUETOOTH_DATA_UPDATE
+
+    coordinator = PassiveBluetoothDataUpdateCoordinator(
+        hass, _LOGGER, "aa:bb:cc:dd:ee:ff", _async_generate_mock_data
+    )
+    saved_callback = None
+
+    def _async_register_callback(_hass, _callback, _matcher):
+        nonlocal saved_callback
+        saved_callback = _callback
+        return lambda: None
+
+    with patch(
+        "homeassistant.components.bluetooth.passive_update_coordinator.async_register_callback",
+        _async_register_callback,
+    ):
+        cancel_coordinator = coordinator.async_setup()
+
+    entity_key = PassiveBluetoothEntityKey("temperature", None)
+    entity_key_events = []
+    all_events = []
+    mock_entity = MagicMock()
+    mock_add_entities = MagicMock()
+
+    def _async_entity_key_listener(data: PassiveBluetoothDataUpdate | None) -> None:
+        """Mock entity key listener."""
+        entity_key_events.append(data)
+
+    cancel_async_add_entity_key_listener = coordinator.async_add_entity_key_listener(
+        _async_entity_key_listener,
+        entity_key,
+    )
+
+    def _all_listener(data: PassiveBluetoothDataUpdate | None) -> None:
+        """Mock an all listener."""
+        all_events.append(data)
+
+    cancel_listener = coordinator.async_add_listener(
+        _all_listener,
+    )
+
+    cancel_async_add_entities_listener = coordinator.async_add_entities_listener(
+        mock_entity,
+        mock_add_entities,
+    )
+
+    saved_callback(GENERIC_BLUETOOTH_SERVICE_INFO, BluetoothChange.ADVERTISEMENT)
+
+    # Each listener should receive the same data
+    # since both match
+    assert len(entity_key_events) == 1
+    assert len(all_events) == 1
+
+    # There should be 4 calls to create entities
+    assert len(mock_entity.mock_calls) == 2
+
+    saved_callback(GENERIC_BLUETOOTH_SERVICE_INFO, BluetoothChange.ADVERTISEMENT)
+
+    # Each listener should receive the same data
+    # since both match
+    assert len(entity_key_events) == 2
+    assert len(all_events) == 2
+
+    # On the second, the entities should already be created
+    # so the mock should not be called again
+    assert len(mock_entity.mock_calls) == 2
+
+    cancel_async_add_entity_key_listener()
+    cancel_listener()
+    cancel_async_add_entities_listener()
+
+    saved_callback(GENERIC_BLUETOOTH_SERVICE_INFO, BluetoothChange.ADVERTISEMENT)
+
+    # Each listener should not trigger any more now
+    # that they were cancelled
+    assert len(entity_key_events) == 2
+    assert len(all_events) == 2
+    assert len(mock_entity.mock_calls) == 2
+
+    cancel_coordinator()
+
+
+GOVEE_B5178_REMOTE_SERVICE_INFO = BluetoothServiceInfo(
```

Comment:
The govee data isn't used.

Reply:
> The govee data isn't used.

It will be in the next test.  Sadly I ran out of time since the A/C went out and required my immediate attention so I never wrote the next test